netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -187,6 +190,55 @@ public AsciiString scheme() {
     }
   }
 
+
+  static final class ProxyNegotiator implements ProtocolNegotiator {
+    private final ProxyHandler proxyHandler;
+    private final ProtocolNegotiator negotiator;
+
+    public ProxyNegotiator(ProxyHandler proxyHandler, ProtocolNegotiator negotiator) {
+      this.negotiator = negotiator;
+      this.proxyHandler = proxyHandler;
+    }
+
+    @Override
+    public Handler newHandler(GrpcHttp2ConnectionHandler http2Handler) {
+      return new WithProxyHandler(proxyHandler, negotiator.newHandler(http2Handler));
+    }
+  }
+
+  static final class WithProxyHandler extends AbstractBufferingHandler
+      implements ProtocolNegotiator.Handler {
+    private ProxyHandler proxyHandler;
+
+    public WithProxyHandler(ProxyHandler proxyHandler, ChannelHandler... handlers) {
+      super(handlers);
+      checkNotNull(proxyHandler, proxyHandler);
+      this.proxyHandler = proxyHandler;
+    }
+
+    @Override
+    public AsciiString scheme() {
+      return Utils.HTTP;
```

Comment:
This needs to delegate to `negotiator`. It should generally be HTTPS, but it's based on the other negotiator used. It's necessary for determining the correct scheme to use on the HTTP/2 connection.